# ChromPolymerDB_tutorial

<!-- ABOUT THE PROJECT -->
The spatial organization of the genome plays a fundamental role in gene regulation, replication, and other biological processes. High-throughput chromosome conformation capture (Hi-C) techniques have advanced our understanding of genome architecture, but they primarily produce 2D, population-averaged data, limiting insights into individual cell chromatin structures. To bridge this gap, this project introduces a computational method and web-based platform designed for high-resolution visualization and analysis of single-cell 3D chromatin conformations.

<img src="image/main_page.png" alt="Main Page" title="Main Page" width="1000" />

<!-- GETTING STARTED -->
## Step 1. Querying the Data

- Our database provides two ways to query the Hi-C data:
  1. Based on the cell line name and genomic location.

     <img src="image/data_query_short.png" alt="data_query" title="data_query" width="1000" />
     
  2. Based on the cell line name and gene name.

     <img src="image/data_query_gene_short.png" alt="data_query" title="data_query" width="1000" />

After entering all the information, click the <code style="color : Cyan">Show Heatmap</code>
<!-- GETTING STARTED -->
## Step 2. Examine the FoldRec interactions

Here we take IMR90 chr8:127,300,000-128,300,000 as an example

<!-- GETTING STARTED -->
## Step 3. Generating the single-cell chromatin conformations
<!-- GETTING STARTED -->
## Step 4. 
